script.js

```diff
@@ -0,0 +1,295 @@
+        </div>`);
+    row.appendChild(key);
+  }
+  keyBoard.appendChild(row);
+}
+
+function addActive(elem) {
+  elem.classList.add('active');
+}
+
+function asincron(func, milliseconds) {
+  let lastCall = 0;
+  return function () {
+    const now = Date.now();
+    if (lastCall + milliseconds < now) {
+      lastCall = now;
+      return func.apply(this, arguments);
+    }
+  };
+}
+
+function removeActive(elem) {
+  elem.classList.remove('active');
+}
+const changeCase = () => {
+  const langElem = keyBoard.querySelectorAll(`div > .${lang}`);
+  console.log(langElem);
+  for (let i = 0; i < langElem.length; i++) {
+    langElem[i].querySelectorAll('span')[0].classList.toggle('hidden');
+    langElem[i].querySelectorAll('span')[1].classList.toggle('hidden');
+  }
+};
+
+const changeLang = () => {
+  const prevLang = keyBoard.querySelectorAll(`div > .${lang}`);
+  for (let i = 0; i < prevLang.length; i++) {
+    prevLang[i].classList.toggle('hidden');
+    prevLang[i].querySelectorAll('span')[0].classList.toggle('hidden');
+  }
+  if (lang === 'rus') {
+    lang = 'eng';
+    localStorage.setItem('lang', lang);
+  } else {
+    lang = 'rus';
+    localStorage.setItem('lang', lang);
+  }
+  const nextLang = keyBoard.querySelectorAll(`div > .${lang}`);
+  for (let i = 0; i < nextLang.length; i += 1) {
+    nextLang[i].classList.toggle('hidden');
+    nextLang[i].querySelectorAll('span')[0].classList.toggle('hidden');
+  }
+};
+
+if (localStorage.lang === 'eng') {
+  changeLang();
+}
+
+document.addEventListener('keydown', asincron((e) => {
+  const elem = keyBoard.getElementsByClassName(e.code)[0];
+
+  if (e.altKey && e.ctrlKey && (e.keyCode === 18 || e.keyCode === 17)) {
+    addActive(elem);
+    changeLang();
+    return false;
+  }
+  switch (e.code) {
+    case 'MetaLeft':
+      break;
+    case 'Tab':
+
+      addActive(elem);
+      textArea.value += '    ';
+      break;
+    case 'Enter':
+
+      addActive(elem);
+      textArea.value += '\n';
+      break;
+    case 'CapsLock':
+      if (capslock) {
+        removeActive(elem);
+        capslock = false;
+      } else {
+        addActive(elem);
+        capslock = true;
+      }
+
+      changeCase();
+      break;
+    case 'Backspace':
+      textArea.value = textArea.value.substr(0, textArea.value.length - 1);
+      addActive(elem);
+      break;
+    case 'Delete':
+
+      addActive(elem);
+      break;
+    case 'AltLeft':
+    case 'AltRight':
+
+      addActive(elem);
+      break;
+    case 'ControlLeft':
+    case 'ControlRight':
+
+      addActive(elem);
+      break;
+    case 'ShiftLeft':
+    case 'ShiftRight':
+
+      addActive(elem);
+      changeCase();
+      break;
+    default:
+      addActive(elem);
+      textArea.value += elem.querySelectorAll(':not(.hidden)')[1].textContent;
+      break;
+  }
+}, 10));
+
+keyBoard.addEventListener('mousedown', (e) => {
+  const elem = e.target.closest('.key');
+  switch (elem.classList[1]) {
```

Comment:
How different this `switch` and `switch` at the line 168??

Reply:
document.addEventListener('keydown' || 'mousedown', asincron((e) => {.....
don't work, i didn't know how to awoid code dublication

Reply:
Very simple: just utilize the same code into separate method and reuse it in both places.

And, yes `document.addEventListener('keydown' || 'mousedown', ` - never will work, because the first argument must be an event type, see https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener